### DISCUSSION TITLE
Set NODE_OPTIONS=--max_old_space_size for Angular app in build target

### DISCUSSION THREAD

**Discussion by GeorgianStan** [nrwl/nx#19577] [General]
I have an Angular application that is built using the `@nx/angular:webpack-browser` executor.

```json
"build": {
  "executor": "@nx/angular:webpack-browser",
...
}
```
This is a large application and requires `NODE_OPTIONS=--max_old_space_size` to be set to a larger value. Until now I used `set` and `export` to set the value of this flag in Windows/Linux and then run the build command.

The first command was

```bash
set NODE_OPTIONS=--max_old_space_size=MY_VALUE`
```
and the second was

```bash
npm run build -- MY_APP
```
which translates to `nx build`.

I would like to get rid of the first command and set the flag from the `project.json` file. I tried with a custom executor and `cross-env' [package](https://www.npmjs.com/package/cross-env), but no luck.

```json
"node-options": {
  "executor": "nx:run-commands",
  "options": {
    "command": "cross-env NODE_OPTIONS=--max_old_space_size=5000"
    }
  },
"build": {
  "executor": "@nx/angular:webpack-browser",
  "dependsOn": ["^build", "node-options"],
...
```
Is there a way to achieve this?

**Comment by kushal-chhetri7** [comment]
yes you can increase the value of max_old_space_size to > 8000 when you start the  local/remote server
ex: export NODE_OPTIONS="--max-old-space-size=8192"
